adagucserverEC/CImageDataWriter.cpp

```diff
@@ -2414,10 +2414,10 @@ int CImageDataWriter::end() {
           CT::string resultJSON;
           if (srvParam->JSONP.length() == 0) {
             CDBDebug("CREATING JSON");
-            printf("%s%c%c\n", "Content-Type: application/json", 13, 10);
+            printf("%s%s%c%c\n", "Content-Type: application/json", srvParam->getCacheControlHeader(CSERVERPARAMS_CACHE_CONTROL_OPTION_SHORTCACHE).c_str(), 13, 10);
           } else {
             CDBDebug("CREATING JSONP %s", srvParam->JSONP.c_str());
-            printf("%s%c%c\n%s(", "Content-Type: application/javascript", 13, 10, srvParam->JSONP.c_str());
+            printf("%s%s%c%c\n", "Content-Type: application/javascript", srvParam->getCacheControlHeader(CSERVERPARAMS_CACHE_CONTROL_OPTION_SHORTCACHE).c_str(), 13, 10);
```

Comment:
The srvParam->JSONP.c_str()) is still needed for the adaguc-viewer

Reply:
Fixed